From 2696bda9f0bd3811b7d51930c5a619dcb0d64d7b Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 27 Sep 2011 16:42:40 -0400
Subject: [PATCH] Fall back to old efi GOP detection behavior.

It appears that on many machines, the GOP driver is not connected to PCI_IO.
The fix to make some macs work better didn't take this into account, so it
broke those machines.  This patch falls back to the older behavior which worked
on those machines.

---
 efi/efigraph.c |    3 +++
 efi/efiuga.c   |    3 +++
 2 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/efi/efigraph.c b/efi/efigraph.c
index c285d39..fc88123 100644
--- a/efi/efigraph.c
+++ b/efi/efigraph.c
@@ -1413,6 +1413,9 @@ enable(struct graphics_backend *backend)
 
 	grub_free(handles);
 
+	if (!eg->output_intf)
+	    eg->output_intf = grub_efi_locate_protocol(&graphics_output_guid, NULL);
+
         if (!eg->output_intf)
             goto fail;
 
diff --git a/efi/efiuga.c b/efi/efiuga.c
index e035c55..b27500b 100644
--- a/efi/efiuga.c
+++ b/efi/efiuga.c
@@ -912,6 +912,9 @@ enable(struct graphics_backend *backend)
 
 	grub_free(handles);
 
+	if (!uga->draw_intf)
+	    uga->draw_intf = grub_efi_locate_protocol(&draw_guid, NULL);
+
         if (!uga->draw_intf) {
             grub_free(uga);
             return 0;
-- 
1.7.6

